mirascope/openai/prompt.py

```diff
@@ -308,22 +337,33 @@ async def async_extract(self, schema, retries=0):
             openai_tool = OpenAITool.from_model(schema)
             return_tool = False
 
-        completion = await self.async_create(tools=[openai_tool])
+        try:
```

Comment:
define self._start_time here? then you can remove 342 and 344 and 328

Reply:
create and async_create also set start time, but that one doesn't take into account all of the time spend converting tools, so I did it this way

Reply:
we are timing the actual completion, not the time it takes to do conversion.

Reply:
Only completion has it now